fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/serialization/LoanProductDataValidator.java

```diff
@@ -744,9 +745,38 @@ public void validateForCreate(final String json) {
         baseDataValidator.reset().parameter(LoanProductConstants.OVER_DUE_DAYS_FOR_REPAYMENT_EVENT).value(overDueDaysForRepaymentEvent)
                 .integerZeroOrGreater();
 
+        if (this.fromApiJsonHelper.parameterExists(LoanProductConstants.ENABLE_DOWN_PAYMENT, element)) {
+            final Boolean enableDownPayment = this.fromApiJsonHelper.extractBooleanNamed(LoanProductConstants.ENABLE_DOWN_PAYMENT, element);
+            baseDataValidator.reset().parameter(LoanProductConstants.ENABLE_DOWN_PAYMENT).value(enableDownPayment).ignoreIfNull()
+                    .validateForBooleanValue();
+            validateDownPaymentPercentage(enableDownPayment, baseDataValidator, element);
+        }
+
         throwExceptionIfValidationWarningsExist(dataValidationErrors);
     }
 
+    private void validateDownPaymentPercentage(Boolean enableDownPayment, DataValidatorBuilder baseDataValidator, JsonElement element) {
+        if (enableDownPayment) {
+            if (this.fromApiJsonHelper.parameterExists(LoanProductConstants.DISBURSED_AMOUNT_PERCENTAGE_DOWN_PAYMENT, element)) {
+                BigDecimal disbursedAmountPercentageDownPayment = this.fromApiJsonHelper
+                        .extractBigDecimalWithLocaleNamed(LoanProductConstants.DISBURSED_AMOUNT_PERCENTAGE_DOWN_PAYMENT, element);
+                baseDataValidator.reset().parameter(LoanProductConstants.DISBURSED_AMOUNT_PERCENTAGE_DOWN_PAYMENT)
+                        .value(disbursedAmountPercentageDownPayment).notLessThanMin(BigDecimal.ONE)
+                        .notGreaterThanMax(BigDecimal.valueOf(100)).scaleNotGreaterThan(Integer.valueOf(6));
```

Comment:
Integer.valueOf(6) can be simplified: 6 ;)

Reply:
done